test/org/javarosa/core/TestBenchmark.java

```diff
@@ -0,0 +1,198 @@
+package org.javarosa.core;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.TimeUnit;
+
+import org.javarosa.core.model.FormDef;
+import org.javarosa.core.model.instance.ExternalDataInstance;
+import org.javarosa.core.model.instance.TreeElement;
+import org.javarosa.core.reference.InvalidReferenceException;
+import org.javarosa.core.reference.ReferenceManager;
+import org.javarosa.core.test.FormParseInit;
+import org.javarosa.form.api.FormEntryController;
+import org.javarosa.xform.parse.FormParserHelper;
+import org.javarosa.xform.parse.XFormParser;
+import org.javarosa.xml.util.InvalidStructureException;
+import org.javarosa.xml.util.UnfullfilledRequirementsException;
+import org.junit.Test;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.results.RunResult;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.TimeValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xmlpull.v1.XmlPullParserException;
+
+
+import static org.javarosa.core.reference.ReferenceManagerTestUtils.buildReferenceFactory;
+import static org.javarosa.test.utils.ResourcePathHelper.r;
+import static org.junit.Assert.fail;
+
+public class TestBenchmark {
```

Comment:
I'd suggest that we have just one `@Benchmark`. Otherwise, I'm forced to run all the benchmarks in the class, or to change the `OptionsBuilder()`.

Reply:
@ggalmazor Do you mean on Benchmark File Per use case ?

Reply:
At least we could try this at the beginning. I fear that we might get too many files, but it's really easier to deal and work with benchmarks that way.